packages/plugins/Scale/src/store/index.ts

```diff
@@ -70,12 +66,45 @@ export const makeStoreModule = () => {
             : `${scaleNumber} m`
         commit('setScaleWithUnit', newValue)
       },
+      zoomToScale(
+        { dispatch, rootGetters: { configuration } },
+        zoomLevel: number
+      ): void {
+        if (configuration?.scale?.zoomMethod) {
+          dispatch(configuration.scale.zoomMethod, zoomLevel, { root: true })
+        }
+      },
     },
     mutations: {
       ...generateSimpleMutations(getInitialState()),
     },
     getters: {
       ...generateSimpleGetters(getInitialState()),
+      zoomOptions: (_, __, ___, rootGetters) => {
+        const options = rootGetters.configuration.options
+
+        if (!options) {
+          console.error(
+            "@polar/plugin-scale: Configuration parameter `options` is missing. Scale switcher won't be rendered."
+          )
+          return []
+        }
+        return options
+      },
+      showScaleSwitcher: (_, getters, ___, rootGetters) => {
+        return (
+          rootGetters.configuration?.scale?.showScaleSwitcher &&
+          getters.zoomOptions.length > 0
+        )
+      },
+      zoomMethod: (_, __, ___, rootGetters) => {
+        return (
+          rootGetters.configuration?.scale?.zoomMethod ||
+          console.error(
+            "@polar/plugin-scale: Configuration parameter `zoomMethod` is missing. Scale switcher won't be rendered."
+          )
+        )
```

Comment:
```suggestion
        const zoomMethod = rootGetters.configuration?.scale?.zoomMethod

        if (!zoomMethod) {
          console.error(
            "@polar/plugin-scale: Configuration parameter `zoomMethod` is missing. Scale switcher won't be rendered."
          )
          return ''
        }
        return zoomMethod
```
This may otherwise lead to the situation that `zoomMethod` may not be a string (contrary to the type definition).